src/common.ts

```diff
@@ -831,3 +831,9 @@ export function classifyIdentifier(parser: ParserState, context: Context, t: Tok
 
   if (!isValidIdentifier(context, t)) report(parser, Errors.Unexpected);
 }
+
+/**
+ * A helper function to check parser state, workaround TS2678 error
+ * @see https://github.com/microsoft/TypeScript/issues/9998
+ */
+export const is = (value: number | boolean) => value;
```

Comment:
It's annoying that TS narrowing down the type of `parser.token` incorrectly, [issue](https://github.com/microsoft/TypeScript/issues/9998), this function prevents that.

Do you have a better idea?

Reply:
I will have a play.

Reply:
Is this a tsc v5 bug? tsc v4 has no such issue, but tsc v4 fails all the places.

Reply:
I would suggest to hold off tsc v5 before we have a proper solution.